### PR TITLE
Fix spell checker

### PR DESCRIPTION
**Pull request by kevinthesun**
Fix spell checker sudo issue.

### CODE REVIEW

**Comment by piiswrong** [comment]
why sudo?

**Comment by kevinthesun** [comment]
Because right now all jenkins jobs have changed user to 'ec2-user', which results in permission issue.

**Comment by piiswrong** [comment]
Can one of the admins verify this patch?

**Comment by kevinthesun** [comment]
@lxn2 This change is for 'ec2-user' user.

File: tests/nightly/TestDoc/doc_spell_grammar.sh [hunk 1]
```diff
@@ -17,18 +17,18 @@ make -j 4 || exit -1
 
 echo "BUILD python2 mxnet"
 cd ./python
-python setup.py install || exit 1
+sudo python setup.py install || exit 1
```
Comment (lxn2) [inline tests/nightly/TestDoc/doc_spell_grammar.sh]: Should be able to remove this

File: tests/nightly/TestDoc/doc_spell_grammar.sh [hunk 2]
```diff
@@ -17,18 +17,18 @@ make -j 4 || exit -1
...
 
 echo "BUILD python3 mxnet"
-python3 setup.py install || exit 1
+sudo python3 setup.py install || exit 1
```
Comment (lxn2) [inline tests/nightly/TestDoc/doc_spell_grammar.sh]: This as well. Then add
`export PYTHONPATH=${PWD}/python`

File: tests/nightly/TestDoc/doc_spell_grammar.sh [hunk 3]
```diff
@@ -17,18 +17,18 @@ make -j 4 || exit -1
...
 
 echo "Install other dependencies"
 cd ..
-yum -y install enchant
-pip install pyenchant
-pip install grammar-check
-pip install html2text
-pip install sphinx==1.5.1 CommonMark==0.5.4 breathe mock==1.0.1 recommonmark
+sudo yum -y install enchant
```
Comment (lxn2) [inline tests/nightly/TestDoc/doc_spell_grammar.sh]: Why aren't these installs in the Dockerfile? They're test dependencies.

File: tests/nightly/TestDoc/doc_spell_grammar.sh [hunk 4]
```diff
@@ -37,9 +37,9 @@ make html
 
 echo "Check spell and grammar for documentation"
 cd ../tests/nightly/TestDoc
-rm -rf web-data
-git clone https://github.com/dmlc/web-data.git
-cp web-data/mxnet/doc/en_US-large.aff web-data/mxnet/doc/en_US-large.dic web-data/mxnet/doc/en_US.aff web-data/mxnet/doc/en_US.dic /usr/share/myspell
-python doc_spell_checker.py
+sudo rm -rf web-data
```
Comment (lxn2) [inline tests/nightly/TestDoc/doc_spell_grammar.sh]: Is sudo actually necessary for these lines?

**Comment by mli** [comment]
please consider put a `Dockerfile.spell_check` under tests/ci_build/. Then you can run the check by `tests/ci_build/ci_build.sh spell_check your_check_command`

we should avoid scripts that contains sudo to change users' systems.

**Pull request closed without merging**